components/masking_functions/include/masking_functions/sql_escape_functions.hpp

```diff
@@ -0,0 +1,29 @@
+/* Copyright (c) 2023 Percona LLC and/or its affiliates. All rights reserved.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; version 2 of the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software Foundation,
+   51 Franklin Street, Suite 500, Boston, MA 02110-1335 USA */
+
+#ifndef MASKING_FUNCTIONS_SQL_ESCAPE_FUNCTIONS_HPP
+#define MASKING_FUNCTIONS_SQL_ESCAPE_FUNCTIONS_HPP
+
+#include <string>
+
+#include "masking_functions/charset_string_fwd.hpp"
+
+namespace masking_functions {
+
+std::string escape_string(const charset_string &cs_str);
```

Comment:
Again, IMO it is worth either to add a header or function description based on one in the commit message.

Also if this function is only used by query_builder class, then perhaps it makes sense to make it private to query_builder.cpp? Unless of course you have big plans for using in some other class or project soon.

Reply:
I extracted this function in a separate file as it is the only one in the `component_masking_functions` code that does not use recommended MySQL services APIs and uses `mysys` function directly. Basically, I wanted to have this `exception` as a separate translation unit. But in general you are right, I should have make this function a private one in the `sql_builder class`. Hopefully, one day Oracle will include it into one of the `mysql_query` services..
Added a corresponding description / explanation.

Reply:
OK. Thanks for explaining!